### PR TITLE
Add BasisTextureLoader example

### PR DESCRIPTION
**Pull request by takahirox**
This PR adds `BasisTextureLoader` example.

Live demo: https://raw.githack.com/takahirox/three.js/BasisTextureLoaderExample/examples/index.html#webgl_loader_texture_basis

`kodim20.basis` is from https://github.com/BinomialLLC/basis_universal/tree/master/webgl/texture/assets

I think this example may be too simple. But better than no example. Users can see the loader works and know the API from the example. Please feel free to replace with a cool and/or interesting one later.

#16524

### CODE REVIEW

File: examples/webgl_loader_texture_basis.html [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+			init();
+			animate();
+
+			function init() {
+
+				renderer = new THREE.WebGLRenderer( { antialias: true } );
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				document.body.appendChild( renderer.domElement );
```
Comment (donmccurdy) [inline examples/webgl_loader_texture_basis.html]: BasisTextureLoader currently marks all textures as sRGB:

https://github.com/mrdoob/three.js/blob/b39de785004e155e744fc47604de5e973135746b/examples/js/loaders/BasisTextureLoader.js#L144-L151

We should probably remove that, and the `flipY=false` preset, which both made sense in the original Basis example but not so much in a general-purpose texture loader.

My suggestion for this PR would be:

- [ ] Remove flipY=false and encoding=sRGBEncoding from the loader
- [ ] Add encoding=sRGBEncoding after loading the texture (in the example itself)
- [ ] set renderer.gammaOutput=true, renderer.gammaFactor=2.2 as best practice.

**Comment by donmccurdy** [comment]
Good to have simple examples, I think!

**Comment by WestLangley** [comment]
Personally, I would just copy the other `webgl_loader_texture_xxx` examples.

https://raw.githack.com/takahirox/three.js/BasisTextureLoaderExample/examples/index.html?q=hdr#webgl_loader_texture_hdr

**Comment by takahirox** [comment]
Thanks for the reviews.

> Personally, I would just copy the other webgl_loader_texture_xxx examples.

The reason why I didn't copy but made a new example rotating cube is I wanted to show transcode doesn't block main thread.

Reply to the other comments will be tomorrow (or later).

File: examples/webgl_loader_texture_basis.html (hunk 1 above)
Comment (takahirox) [reply examples/webgl_loader_texture_basis.html]: > Remove flipY = false from the loader

`flipY = true` doesn't seem to work for compressed texture and default `CompressedTexture.flipY` is `false`. Removing from the loader looks right direction. Updated.

> Move encoding = sRGBEncoding into example

Updated.

> renderer.gammaOutput = true, renderer.gammaFactor=2.2

Thanks, I missed them. Updated.

**Comment by mrdoob** [comment]
Screen Shot 2019-05-27 at 7 37 46 PM

Is this expected?

**Comment by takahirox** [comment]
No. And I can't reproduce here... The root issue seems the server 504 error? Does it still happen even if you reload?

**Comment by mrdoob** [comment]
Weird, I can no longer reproduce...

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**